Document integer size limit for number_input and slider

MathCatsAnd opened the pull request:
Added note to docstrings for `number_input` and `slider` to clarify limitations for storing and returning large integers. Closes streamlit/docs#106

## 📚 Context
Programmatic limits exists for arguments in `st.number_input` and `st.slider` to prevent creating widgets having values larger in magnitude than `+/- (1<<53) -1`. These limits can be silently encountered if not passed as an argument upon creation. In that instance, incorrect integers will be stored and returned. Per a stale issue in streamlit/docs, a note has been added to the docstrings.

- What kind of change does this PR introduce?
  - [X] Documentation

**Contribution License Agreement**
By submitting this pull request you agree that all contributions to this project are made under the Apache 2.0 license.

MathCatsAnd committed:
Integer size limit for number_input and slider

Added note to docstrings for `number_input` and `slider` to clarify limitations for storing and returning large integers

vdonato reviewed (commented):
Thanks @MathCatsAnd! This LGTM, but I'll let @snehankekre weigh in for an official review as our authority on docs 🙂

MathCatsAnd commented:
I just noticed a capitalization error from copy-pasting-editing. I will modify Javascript -> JavaScript to fix it. 🙂

MathCatsAnd committed:
Typesetting

Changed Javascript to JavaScript for correct capitalization.

snehankekre reviewed (approved):
Approving pending comment.

Thank you for fixing this long standing issue, @MathCatsAnd! 😄 Your note explains the problem, the cause, and the implications in a clear and organized way. nit: it could be slightly more concise. How about the following?

```
Integer values exceeding ``+/- (1<<53) - 1`` cannot be accurately stored or returned by the widget due to serialization constraints between the Python server and JavaScript client. Such numbers must be handled as floats, leading to a loss in precision.
```

Note: [inline code](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#inline-markup) in reStructuredText should be enclosed with two backquotes.

MathCatsAnd commented:
To Nit myself:

Since `(1<<53) - 1` is a valid command but `+/-` isn't exactly, by any chance would it be more proper pull the +/- out like this:
```Integer values exceeding +/- ``(1<<53) - 1`` cannot be ```...

And would it be better to direct second sentence to "you" to try and be even clearer that the widget isn't going to convert automatically on the user's behalf? I had already tried to consider this, but I thought this might be even better. After all, the way things can go wrong go beyond just rounding error, so... ```You must handle such numbers as floats....```

```Integer values exceeding +/- ``(1<<53) - 1`` cannot be accurately
stored or returned by the widget due to serialization constraints
between the Python server and JavaScript client. You must handle
such numbers as floats, leading to a loss in precision.```

snehankekre commented:
Yup, both of those look like improvements 👍

MathCatsAnd committed:
Docstring edit for succinctness -- integer limits

Slightly reworded and trimmer warning about integer size limits in the docstrings of `slider` and `number_input`

MathCatsAnd commented:
Would it also be worth putting in an issue on streamlit/streamlit to request having `None` limits automatically interpretted as `min_value=-2**53+1, max_value=2**53-1,` when in integer mode? The widget behaves nicely, producing the correct warning to the user when manually set like that so that might help avoid giving some users a surprise:

![image](https://github.com/streamlit/streamlit/assets/98661771/b9c8854d-395b-4349-ae16-0f8f9ac5be6d)

vdonato commented:
> Would it also be worth putting in an issue on streamlit/streamlit to request having None limits automatically interpretted as min_value=-2**53+1, max_value=2**53-1, when in integer mode?

I think so! I'll go ahead and file an enhancement request for this. The one thing that may be slightly awkward about doing this by default is that the resulting numbers in the error message would probably look entirely arbitrary to everyone except for those that know what JS's min/max int values are. This is almost definitely still better than over/underflowing silently, but there may be a better product solution that we're not considering.
